content/id/docs/tasks/administer-cluster/highly-available-master.md

```diff
@@ -0,0 +1,177 @@
+
+* `KUBE_GCE_ZONE=zone` - zona dimana replika master akan dihapus.
+
+* `KUBE_REPLICA_NAME=replica_name` - (opsional) nama replika master yang dihapus.
+Jika kosong: replika mana saja dari zona yang diberikan akan dihapus.
+
+Contoh perintah berikut menghapus replika master dari klaster HA yang sudah ada sebelumnya:
+
+```shell
+KUBE_DELETE_NODES=false KUBE_GCE_ZONE=europe-west1-c ./cluster/kube-down.sh
+```
+
+## Mengatasi replika master yang gagal
+
+Jika salah satu replika master di klaster HA kamu gagal,
+praktek terbaik adalah menghapus replika dari klaster kamu dan menambahkan replika baru pada zona yang sama.
+Contoh perintah dibawah menunjukkan proses berikut ini:
+
+1. Menghapus replika yang gagal:
+
+```shell
+KUBE_DELETE_NODES=false KUBE_GCE_ZONE=replica_zone KUBE_REPLICA_NAME=replica_name ./cluster/kube-down.sh
+```
+
+2. Menambahkan replika baru untuk menggantikan replika yang lama
+
+```shell
+KUBE_GCE_ZONE=replica-zone KUBE_REPLICATE_EXISTING_MASTER=true ./cluster/kube-up.sh
+```
+
+## Praktek terbaik untuk mereplikasi master untuk klaster HA
+
+* Cobalah untuk menempatkan replika master pada zona yang berbeda. Pada saat terjadi kegagalan zona, semua master yang ditempatkan dalam zona tersebut akan gagal.
+Untuk bertahan dari kegagalan zona, tempatkan juga Node pada beberapa zona yang lain
+(Lihatlah [multi-zona](/id/docs/setup/best-practices/multiple-zones/) untuk lebih detail).
+
+* Jangan gunakan klaster dengan dua replika master. Konsensus pada klaster dengan dua replika membutuhkan kedua replika tersebut berjalan pada saat mengubah keadaan yang persisten.
+Akibatnya, kedua replika tersebut diperlukan dan kegagalan salah satu replika mana pun mengubah klaster dalam status kegagalan mayoritas.
+Dengan demikian klaster dengan dua replika lebih rendah keadaannya, dalam hal HA, daripada klaster dengan replika tunggal.
+
+* Ketika kamu menambahkan sebuah replika master, status klaster (etcd) disalin ke sebuah _instance_ baru.
+Jika klaster itu besar, mungkin butuh waktu yang lama untuk menduplikasi keadaannya.
+Operasi ini dapat dipercepat dengan memigrasi direktori data etcd, seperti yang dijelaskan [disini](https://coreos.com/etcd/docs/latest/admin_guide.html#member-migration)
+(Kita sedang mempertimbangkan untuk menambahkan dukungan untuk migrasi direktori data etcd di masa mendatang).
+
+
+
+<!-- discussion -->
+
+## Catatan implementasi
+
+![ha-master-gce](/images/docs/ha-master-gce.png)
+
+### Ikhtisar
+
+Setiap replika master akan menjalankan komponen berikut dalam mode berikut:
+
+* _instance_ etcd: semua _instance_ akan dikelompokkan bersama menggunakan konsensus;
+
+* server API : setiap server akan berbicara dengan lokal etcd - semua server API pada cluster akan tersedia;
+
+* pengontrol (_controller_), penjadwal (_scheduler_), dan _scaler_ klaster automatis: akan menggunakan mekanisme sewa - dimana hanya satu _instance_ dari masing-masing mereka yang akan aktif dalam klaster;
```

Comment:
```suggestion
* pengontrol (_controller_), penjadwal (_scheduler_), dan _scaler_ klaster otomatis: akan menggunakan mekanisme sewa - dimana hanya satu _instance_ dari masing-masing mereka yang akan aktif dalam klaster;
```

Reply:
"automatis" is more official. 
https://kbbi.kemdikbud.go.id/entri/automatis

Reply:
> "automatis" is more official.
> https://kbbi.kemdikbud.go.id/entri/automatis

If you open https://kbbi.kemdikbud.go.id/entri/otomatis the "automatis" listed as tidak baku